Sources/Defaults/Defaults+iCloud.swift

```diff
@@ -0,0 +1,462 @@
+			let remoteTimestamp = remoteStorage.object(forKey: defaultsSyncKey) as? Date
+		else {
+			return
+		}
+
+		if
+			let localTimestamp = localStorage.object(forKey: defaultsSyncKey) as? Date,
+			localTimestamp > remoteTimestamp
+		{
+			return
+		}
+
+		for key in self.keys where changedKeys.contains(key.name) {
+			backgroundQueue.async {
+				await self.syncKey(key, .remote)
+			}
+		}
+	}
+}
+
+/**
+`Defaults.iCloud` logging related functions.
+*/
+extension Defaults.iCloudSynchronizer {
+	@available(macOS 11, iOS 14, tvOS 14, watchOS 7, *)
+	private static let logger = Logger(OSLog.default)
+
+	private static func logKeySyncStatus(_ key: Defaults.Keys, source: DataSource, syncStatus: SyncStatus, value: Any? = nil) {
+		guard Defaults.iCloud.isDebug else {
+			return
+		}
+		var destination: String
+		switch source {
+		case .local:
+			destination = "from local"
+		case .remote:
+			destination = "from remote"
+		}
+		var status: String
+		var valueDescription = " "
+		switch syncStatus {
+		case .start:
+			status = "Start synchronization"
+		case .isSyncing:
+			status = "Synchronizing"
+			valueDescription = " with value \(value ?? "nil") "
+		case .finish:
+			status = "Finish synchronization"
+		}
+		let message = "\(status) key '\(key.name)'\(valueDescription)\(destination)"
+
+		log(message)
+	}
+
+	private static func log(_ message: String) {
+		guard Defaults.iCloud.isDebug else {
+			return
+		}
+
+		if #available(macOS 11, iOS 14, tvOS 14, watchOS 7, *) {
+			logger.debug("[Defaults.iCloud] \(message)")
+		} else {
+#if canImport(OSLog)
+			os_log(.debug, log: .default, "[Defaults.iCloud] %@", message)
+#else
+			let dateFormatter = DateFormatter()
+			dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss.SSSSSSZZZ"
+			let dateString = dateFormatter.string(from: Date())
+			let processName = ProcessInfo.processInfo.processName
+			let processIdentifier = ProcessInfo.processInfo.processIdentifier
+			var threadID: UInt64 = 0
+			pthread_threadid_np(nil, &threadID)
+			print("\(dateString) \(processName)[\(processIdentifier):\(threadID)] [Defaults.iCloud] \(message)")
+#endif
+		}
+	}
+}
+
+extension Defaults {
+	/**
+	Automatically create synchronization tasks when the added `keys` undergo changed.
+
+	The synchronization task will be created in three different ways.
+
+	1. UserDefaults changes.
+	2. Receive `NSUbiquitousKeyValueStore.didChangeExternallyNotification`.
+	3. Call `syncWithoutWaiting()`.
+
+	> Tip: Using `await sync()` to make sure all synchronization tasks are done.
+
+	```swift
+	let quality = Defaults.Key<Int>("quality", default: 0, iCloud: true)
+	Defaults[quality] = 1
+	await Defaults.iCloud.sync()
+	print(NSUbiquitousKeyValueStore.default.object(forKey: quality.name)) //=> 1
+	```
+	*/
+	public enum iCloud {
+		/**
+		The singleton for Defaults's iCloudSynchronizer.
+		*/
+		static var shared = Defaults.iCloudSynchronizer(remoteStorage: NSUbiquitousKeyValueStore.default)
```

Comment:
The name is not correct. It makes it sound like it's the singleton of `iCloud`.
```suggestion
		static var synchronizer = Defaults.iCloudSynchronizer(remoteStorage: NSUbiquitousKeyValueStore.default)
```